test/battle/ai/ai_switching.c

```diff
@@ -496,3 +496,65 @@ AI_SINGLE_BATTLE_TEST("AI_FLAG_SMART_SWITCHING: AI will switch out if it can't d
         TURN { MOVE(player, MOVE_TACKLE) ; EXPECT_SWITCH(opponent, 1); }
     }
 }
+
+AI_SINGLE_BATTLE_TEST("AI_FLAG_SMART_SWITCHING: AI will switch out if it has been Toxic'd for at least two turns 50% of the time with more than 1/3 HP remaining")
```

Comment:
Not a test review but does this make sense from a player perspective? If you are poisoned you will likely die anyways which means you might lose hp on the incoming mon.

Reply:
Personally I find it questionable at best lol

I'm not responsible for getting this functionality added in the first place and I didn't want to scope this PR to be functional behavioral changes.

I'm down to change or remove this behaviour, but I'd prefer it in an isolated PR. I doubt it'd end up in #5466 either as that's also not focused on functionality changes. We can make an issue to take a look at it later if you like, most of my efforts atm are going towards enabling prediction.

Reply:
If we want to improve the AI then I would say we should remove it because I it is arguably objectively bad to switch in this case.

Reply:
Yeah I'm fully on board. I'll make an issue so I don't forget.